fontforgeexe/fontview.c

```diff
@@ -245,7 +245,71 @@ extern BDFFont *FVSplineFontPieceMeal(SplineFont *sf, int layer, int ptsize, int
     return new;
 }
 
-static void FVDrawGlyph(GWindow pixmap, FontView *fv, int index, int request_expose ) {
+void FVDrawOutlineOnly(GWindow pixmap, SplinePointList *set,
+                       Color fg, enum outlinesfm_flags strokeFillMode, float xoff, float yoff, double scale) {
+	if (!set) {
+		return;
+	}
+	for (SplinePointList *spl = set; spl != NULL; spl = spl->next) {
```

Comment:
It looks like this code mostly duplicates a part of `CVDrawSplineSetOutlineOnly()`. Could you please extract it to a common function?

Reply:
you right i get it from `CVDrawSplineSetOutlineOnly()`

for some reason `CVDrawSplineSetOutlineOnly()` code needs extra attributes. And Y coordinates are inverse. Also a grid drawing code. 

`cy2 = -tab->yoff + cv->height - cy2*tab->scale - dy;`
vs
`cy2 = yoff - spline->to->prevcp.y * scale - dy;`

When i trying to extract function from  `CVDrawSplineSetOutlineOnly()` it has extra dependencies and functions i don't have in FV scope.
`void PixmapDrawSplineSet(CharView *cv, GWindow pixmap, SplinePointList *set, Color fg, enum outlinesfm_flags strokeFillMode, SplinePointList *spl, int activelayer, CharViewTab *tab)`
I was not able to find this in FV scope:
`CharView *cv, int activelayer, CharViewTab *tab`

With my current knowledge of code i wont risk refactor existing code.
If you insist, I will.

Reply:
My intent was to extract only the lines 1533-1586 from CVDrawSplineSetOutlineOnly(), and this is effectively what you did.
I think you can extract them to a function like
```
CVDrawSplinePointList(CharView *cv, GWindow pixmap, SplinePointList *set,
    Color fg, enum outlinesfm_flags strokeFillMode,
    float xoff, float yoff, real scale, int height)
```

You would be able to call that function both from CVDrawSplineSetOutlineOnly() and from font view with appropriate values for xoff/yoff/scale, and from fontview with `cv=NULL, height=0`. 

`rpt()` would need to be modified to NULL-check the `cv` argument. My only concern is about `rint()` calls you removed. Did you do that on purpose?

Reply:
> My only concern is about rint() calls you removed. Did you do that on purpose?
I was trying to remove as much deps as possible from function. In order to make it work. 
I didn't understood what rint() and rpt() are for. So i will look into it.

I need couple days to make it. Will notify then done.

Reply:
@iorsh, done